### PR TITLE
[PT FE] Inverse Operation

### PR DESCRIPTION
**Pull request by PiotrKrzem**
!!! To be merged after [this PR](https://github.com/openvinotoolkit/openvino/pull/22649) !!!

Files concerned in this PR are only the following:
`src/frontends/pytorch/src/op/inverse.cpp`
`src/frontends/pytorch/src/op_table.cpp`
`tests/layer_tests/pytorch_tests/test_inverse.py`

### Details:
 - Add Inverse operation to PyTorch frontend

### Tickets:
 - 131644

### CODE REVIEW

File: src/frontends/pytorch/src/op/inverse.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    if (input_type == element::bf16 || input_type == element::f16 || input_type == element::f32) {
+        result = context.mark_node(std::make_shared<ov::op::v14::Inverse>(input_tensor, false));
+    } else {
+        const auto converted_input =
+            context.mark_node(std::make_shared<ov::op::v0::Convert>(input_tensor, element::f32));
+        auto inverse = context.mark_node(std::make_shared<ov::op::v14::Inverse>(converted_input, false));
+        result = context.mark_node(std::make_shared<ov::op::v0::Convert>(inverse, input_type));
+    }
+
+    if (!context.input_is_none(2)) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/inverse.cpp:32]: I suspect this meant to be index 1

File: src/frontends/pytorch/src/op/inverse.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+OutputVector translate_inverse(const NodeContext& context) {
+    // aten::inverse(Tensor self) -> Tensor
+    // aten::inverse(Tensor self, *, Tensor out) -> Tensor
+    num_inputs_check(context, 1, 2);
+    const auto input_tensor = context.get_input(0);
+    const auto input_type = input_tensor.get_element_type();
+
+    ov::Output<ov::Node> result;
+    if (input_type == element::bf16 || input_type == element::f16 || input_type == element::f32) {
```
Comment (eaidova) [inline src/frontends/pytorch/src/op/inverse.cpp:23]: shoud checking input_dtype.is_real() enough here?

File: src/bindings/python/src/openvino/runtime/opset14/ops.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Factory functions for ops added to openvino opset14."""
+from functools import partial
+from typing import Literal, Optional, Union
+import logging
+
+import numpy as np
```
Comment (eaidova) [inline src/bindings/python/src/openvino/runtime/opset14/ops.py:10]: I do not see numpy usage? is it really should be imported?

File: src/frontends/pytorch/src/op/inverse.cpp (hunk 2 above)
Comment (PiotrKrzem) [reply src/frontends/pytorch/src/op/inverse.cpp:23]: As of now, CPU does not support f64 explicitly, so it is also converted to f32 in order to preserve low latency on inference, hence this manual type check.

File: src/bindings/python/src/openvino/runtime/opset14/ops.py (hunk 3 above)
Comment (PiotrKrzem) [reply src/bindings/python/src/openvino/runtime/opset14/ops.py:10]: I've kept it as it is present in every other opset file, but I can remove it in a followup PR with micro fixes

**Review by mmikolajcz — APPROVED** [review]
For torch FE part, LGTM

**Review by eaidova — APPROVED** [review]

**Pull request merged**